core/corehttp/gateway_handler_unixfs.go

```diff
@@ -17,8 +19,14 @@ import (
 func (i *gatewayHandler) serveUnixFs(w http.ResponseWriter, r *http.Request, resolvedPath ipath.Resolved, contentPath ipath.Path, begin time.Time, logger *zap.SugaredLogger) {
 	ctx, span := tracing.Span(r.Context(), "Gateway", "ServeUnixFs", trace.WithAttributes(attribute.String("path", resolvedPath.String())))
 	defer span.End()
+
+	// Decouple (potential) directory context to cancel it in case it's too big.
+	directoryContext, cancelDirContext := context.WithCancel(ctx)
```

Comment:
nit: we keep the context name in other places, lets to the same here
```suggestion
	// Decouple (potential) directory context to cancel it in case it's too big.
	ctx, cancel := context.WithCancel(ctx)
```